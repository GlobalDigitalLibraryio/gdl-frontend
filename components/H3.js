// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import styled from 'react-emotion';
import media from '../style/media';
import { fonts } from '../style/theme';

const H3 = styled.h3`
  &:first-child {
    margin-top: 0;
  }
  text-transform: uppercase;
  font-weight: ${fonts.weight.medium};
  font-size: 16px;
  line-height: 22px;
  ${media.tablet`
    font-size: 20px;
    line-height: 26px;
  `};
`;

export default H3;
