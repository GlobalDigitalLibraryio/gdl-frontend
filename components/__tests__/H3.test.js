// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createSerializer } from 'jest-emotion';
import * as emotion from 'emotion';
import H3 from '../H3';

expect.addSnapshotSerializer(createSerializer(emotion));

test('Renders', () => {
  const tree = shallow(<H3>Header 3</H3>);

  expect(toJson(tree)).toMatchSnapshot();
});
