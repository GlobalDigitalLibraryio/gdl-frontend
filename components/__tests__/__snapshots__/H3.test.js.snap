// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
.emotion-0 {
  text-transform: uppercase;
  font-weight: 500;
  font-size: 16px;
  line-height: 22px;
}

.emotion-0:first-child {
  margin-top: 0;
}

@media (min-width:768px) {
  .emotion-0 {
    font-size: 20px;
    line-height: 26px;
  }
}

<h3
  className="emotion-0 emotion-1"
>
  Header 3
</h3>
`;
